src/coreclr/tools/Common/Compiler/TypeParser.cs

```diff
@@ -0,0 +1,60 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Internal.TypeSystem;
+using System.Diagnostics;
+
+namespace ILCompiler
+{
+    public class TypeParser
+    {
+        public static TypeDesc GetType(ModuleDesc assembly, string fullName, bool throwIfNotFound)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(fullName));
+            var position = fullName.IndexOf('/');
+            if (position > 0)
+                return GetNestedType(assembly, fullName, throwIfNotFound);
+            string @namespace, name;
+            SplitFullName(fullName, out @namespace, out name);
+
+            return assembly.GetType(@namespace, name, throwIfNotFound);
+        }
+
+        private static MetadataType GetNestedType(ModuleDesc assembly, string fullName, bool throwIfNotFound)
+        {
+            var names = fullName.Split('/');
+            var type = GetType(assembly, names[0], throwIfNotFound);
+
+            if (type == null)
+                return null;
+
+            MetadataType typeReference = (MetadataType)type;
+            for (int i = 1; i < names.Length; i++)
+            {
+                var nested_type = typeReference.GetNestedType(names[i]);
+                if (nested_type == null)
```

Comment:
This doesn't respect the `throwIfNotFound`. I wonder if we need that behavior. Maybe it would be better to delete that parameter and use implied `throwIfNotFound = false` at the point where we need it.